src/pybind/mgr/dashboard/frontend/src/app/ceph/block/rbd-list/rbd-list.component.ts

```diff
@@ -620,14 +630,25 @@ export class RbdListComponent extends ListWithDetails implements OnInit {
     }, []);
   }
 
+  getCopyDisableDesc(selection:CdTableSelection):string | boolean { 
+        const first = selection.first();
+        if( first && first.mirror_mode === 'Disabled'){
+          return `Mirroring needs to be enabled on the image to perform this action`
+        }
+
+        return this.getRemovingStatusDesc(selection) || this.getInvalidNameDisable(selection) || !!selection.first().cdExecuting || selection.first().mirror_mode === 'Disabled';
+
+  }
   getDeleteDisableDesc(selection: CdTableSelection): string | boolean {
     const first = selection.first();
 
     if (first && this.hasClonedSnapshots(first)) {
-      return $localize`This RBD has cloned snapshots. Please delete related RBDs before deleting this RBD.`;
+      return $localize`This RBD has cloned snapshots. Please delete related RBDs before deleting this RBD`;
     }
-
-    return this.getInvalidNameDisable(selection) || this.hasClonedSnapshots(selection.first());
+    if (first && first.primary === false) {
+      return $localize`The image cannot be deleted as it is not primary`;
+    }
+    return this.getInvalidNameDisable(selection) || this.hasClonedSnapshots(selection.first()) || first.primary === false;
```

Comment:
no need for repetition right, its getting checked right above anyways
```suggestion
    return this.getInvalidNameDisable(selection) || this.hasClonedSnapshots(selection.first());
```

Reply:
So @nizamial09 648: is for adding the tooltip and 651 is to disable it.